lib/dest/write-contents/write-symbolic-link.js

```diff
@@ -1,18 +1,44 @@
 'use strict';
 
-var fs = require('graceful-fs');
+var path = require('path');
 
 var fo = require('../../file-operations');
 
 function writeSymbolicLink(file, optResolver, onWritten) {
-  // TODO handle symlinks properly
-  fs.symlink(file.symlink, file.path, function(symlinkErr) {
-    if (fo.isFatalOverwriteError(symlinkErr)) {
-      return onWritten(symlinkErr);
-    }
-
-    onWritten();
-  });
+  var srcPath = file.symlink;
+
+  var isDirectory = file.isDirectory();
```

Comment:
Sorry, my remark wasn't very clear :-)

I agree about the loose coupling, but my point here is that I think the `isDirectory` call here is _wrong_ in the common case of getting a symbolic vinyl as produced by `src()` with `resolveSymlinks` set to false!

Because it is looking at the wrong stats in that case, namely those of the link and not the target, and then `isDirectory` will return false even though the target might be a directory, and as a consequence the symlink type should've been`'junction'` but now won't be.

Does that help?

If you change the test for this, or add a new one, to work with an actually sourced symlink rather than one with a mock `isDirectory()`